src/control/drpc/module_svc.go

```diff
@@ -119,9 +119,18 @@ func (r *ModuleService) ProcessMessage(session *Session, msgBytes []byte) ([]byt
 		err = errors.Errorf("Attempted to call unregistered module")
 		return marshalResponse(msg.GetSequence(), Status_UNKNOWN_MODULE, nil)
 	}
-	respBody, err := module.HandleCall(session, msg.GetMethod(), msg.GetBody())
+	method, ok := GetMethod(module.ID(), msg.GetMethod())
+	if !ok {
+		err = errors.Errorf("Attempted to call unknown module")
+		return marshalResponse(msg.GetSequence(), Status_UNKNOWN_MODULE, nil)
+	}
```

Comment:
Hmm. Either this or the previous module check is redundant. We've now got two places to maintain a list of valid modules (r.modules[] and the switch in `GetMethod()`). Maybe the right way to handle this is to make `GetMethod()` a method on a baseModule struct which is embedded in the the various module types. Then `GetMethod()` only has to validate the method and not the module (which is provided by the embedding struct).

Reply:
Thinking about it some more, I guess that doesn't really save us much. There still need to be a switch on module. Either that, or each module would have to implement its own GetMethod() and that doesn't help much.